### PR TITLE
replace wiht policy method

### PR DESCRIPTION
**Pull request by Akira-Taniguchi**
# Description

#62 
#63 

# Why

Compatible with white paper v2

### CODE REVIEW

File: contracts/Allocator.sol [hunk 1]
```diff
@@ -61,24 +61,31 @@ contract Allocator is Killable, Ownable, UsingConfig, Withdrawable {
...
+		uint256 totalIssuedMetrics = MetricsGroup(config().metricsGroup())
+			.totalIssuedMetrics();
+		uint256 share = market.issuedMetrics() / totalIssuedMetrics;
 		uint256 period = block.number -
 			lastAllocationBlockEachMetrics[_metrics];
-		uint256 allocationPerBlock = _value / period;
+		uint256 lockupValue = lockup.getTokenValue(
+			msg.sender,
+			metrics.property()
+		);
```
Comment (aggre) [inline contracts/Allocator.sol]: In here, lockupValue should indicate the total lockup amount of this property.

Comment (aggre) [reply contracts/Allocator.sol]: This expression indicates how much `msg.sender` (a Market) is locking up to this Property.

File: contracts/Allocator.sol [hunk 2]
```diff
@@ -61,24 +61,30 @@ contract Allocator is Killable, Ownable, UsingConfig, Withdrawable {
 			pendingIncrements[_metrics] == true,
 			"Not asking for an indicator"
 		);
-		address property = metrics.property();
-		uint256 share = market.issuedMetrics() /
-			MetricsGroup(config().metricsGroup()).totalIssuedMetrics();
+		Policy policy = Policy(config().policy());
+		Lockup lockup = Lockup(config().lockup());
+		uint256 totalIssuedMetrics = MetricsGroup(config().metricsGroup())
+			.totalIssuedMetrics();
```
Comment (aggre) [inline contracts/Allocator.sol]: In here, `totalIssuedMetrics` should indicate number of assets in the entire protocol.

File: contracts/libs/Utils.sol [hunk 3]
```diff
@@ -24,3 +24,30 @@ contract AddressSet {
...
+	function get(address _address) public view returns (uint256) {
+		return _addressValueMap[_address];
+	}
+
+	function getSumAllValue() public view returns (uint256) {
+		uint256 sum;
+		for (uint256 i = 0; i < _addresses.length; i++) {
+			sum += _addressValueMap[_addresses[i]];
+		}
+		return sum;
```
Comment (aggre) [inline contracts/libs/Utils.sol]: This code is going to be computationally expensive. There is a risk because so maybe not being able to pay for gas.

Comment (aggre) [reply contracts/libs/Utils.sol]: It's a may fool way, but what about creating a value that gets added each time who runs `add` method and subtracted every time who runs ` remove`(yet not there is) method?

Comment (Akira-Taniguchi) [reply contracts/libs/Utils.sol]: I think so too.
Since there is no remove method yet, only the current addition is performed.

@aggre

**Pull request merged**